### PR TITLE
[Optional Reference] LPS-185137 Convert the Optional Dynamic Greedy references to static references using Snapshot in module dynamic-data-lists-web

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:startup

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by W2jx** [comment]
no conflicts(5/23 8:20)

File: modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/portlet/DDLDisplayPortlet.java [hunk 1]
```diff
@@ -202,17 +203,15 @@ protected void setDDLRecordSetRequestAttribute(RenderRequest renderRequest)
...
-	}
-
 	private static final String _ALIAS = "dynamic-data-list";
 
 	private static final Log _log = LogFactoryUtil.getLog(
 		DDLDisplayPortlet.class);
 
+	private static final Snapshot<DDLWebConfigurationActivator>
+		_ddlWebConfigurationActivatorSnapshot = new Snapshot<>(
+			DDLDisplayPortlet.class, DDLWebConfigurationActivator.class);
```
Comment (dantewang) [inline modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/portlet/DDLDisplayPortlet.java:213]: @W2jx For `Dynamic, Greedy` references, please use 
```
	public Snapshot(
		Class<?> holderClass, Class<T> serviceClass, String filterString,
		boolean dynamic) {
```
and pass `true`.

**Comment by dantewang** [comment]
@W2jx Please also double check your other PRs, thanks!

File: modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/portlet/DDLDisplayPortlet.java (hunk 1 above)
Comment (dantewang) [reply modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/portlet/DDLDisplayPortlet.java:213]: cc @JanisZhang @LisztLiszt @slnn

**Pull request closed without merging**